crates/wasm/src/build.rs

```diff
@@ -0,0 +1,44 @@
+use crate::error::WasmResult;
+use std::str::FromStr;
+use wasm_bindgen::prelude::wasm_bindgen;
+use wasm_bindgen::JsValue;
+
+use crate::utils;
+use penumbra_keys::FullViewingKey;
+use penumbra_proto::core::transaction::v1alpha1 as pb;
+use penumbra_transaction::{plan::ActionPlan, plan::TransactionPlan, WitnessData};
+
+/// Builds a planned [`Action`] specified by
+/// the [`ActionPlan`] in a [`TransactionPlan`].
+/// Arguments:
+///     transaction_plan: `TransactionPlan`
+///     action_plan: `ActionPlan`
+///     full_viewing_key: `bech32m String`,
+///     witness_data: `WitnessData``
+/// Returns: `Action`
+#[wasm_bindgen]
+pub fn build_action(
+    transaction_plan: JsValue,
+    action_plan: JsValue,
+    full_viewing_key: &str,
+    witness_data: JsValue,
+) -> WasmResult<JsValue> {
+    utils::set_panic_hook();
+
+    let transaction_plan: TransactionPlan =
+        serde_wasm_bindgen::from_value(transaction_plan.clone())?;
+
+    let witness_data_proto: pb::WitnessData = serde_wasm_bindgen::from_value(witness_data)?;
+    let witness_data: WitnessData = witness_data_proto.try_into()?;
```

Comment:
Can we just annotate `WitnessData` so we get the auto-try-into stuff?

```rust
#[derive(Clone, Debug, Serialize, Deserialize)]
#[serde(try_from = "pb::WitnessData", into = "pb::WitnessData")]
pub struct WitnessData {
```

Reply:
We can, references https://github.com/penumbra-zone/penumbra/issues/3400. 

I modified the witness as `let witness: WitnessData = serde_wasm_bindgen::from_value(witness_data)?`